packages/calypso-e2e/src/jest-playwright-config/environment.ts

```diff
@@ -374,4 +282,123 @@ class JestEnvironmentPlaywright extends NodeEnvironment {
 	}
 }
 
+/**
+ * Determine the browser to be used for the spec.
+ *
+ * By default, the browser used is defined in the BROWSER_NAME
+ * environment variable, but at times it may be required to
+ * run tests in a different browser.
+ *
+ * To specify a different browser than default,
+ * use the @browser tag in the docblock.
+ *
+ * Example:
+ *
+ * 	`@browser firefox`
+ */
+async function determineBrowser( testFilePath: string ): Promise< BrowserType > {
+	const parsed = parseDocBlock( await fs.readFile( testFilePath, 'utf8' ) );
+	const defaultBrowser = env.BROWSER_NAME;
+
+	// Parsed docblock can return any one of the following:
+	// 	- undefined: if a tag was not found.
+	// 	- single string: if only one instance of a tag was found.
+	//	- string array: if multiple instances of the tag was found.
+	// In this case, we only want to throw if the tag has been defined
+	// multiple times.
+	if ( parsed.browser && typeof parsed.browser !== 'string' ) {
+		throw new Error( 'Multiple browsers defined in docblock.' );
+	}
+
+	// If the browser tag was found in the docblock, look for the
+	// matches in the supported browsers list.
+	// If the browser tag was **not** found, then match based on
+	// the default broswser specified in the BROWSER_NAME
+	// environment variable.
+	const match = supportedBrowsers.find( ( browser ) => {
+		return parsed.browser === undefined
+			? browser.name() === defaultBrowser.toLowerCase()
+			: browser.name() === parsed.browser;
+	} );
+
+	if ( ! match ) {
+		throw new Error( 'Unsupported browser defined in docblock.' );
+	}
+	return match;
```

Comment:
Suggestion for better clarity/error handling on this bit:

```suggestion
		if ( parsed.browser === undefined ) {
			if ( ! env.BROWSER_NAME ) {
				throw new Error( 'Default browser not defined.' );
			}

			const fromEnvVariables = supportedBrowsers.find(
				( browser ) => browser.name().toLowerCase() === env.BROWSER_NAME.toLowerCase()
			);

			if ( ! fromEnvVariables ) {
				throw new Error( `Unsupported default browser: ${ env.BROWSER_NAME }` );
			}

			return fromEnvVariables;
		} else if ( typeof parsed.browser === 'string' ) {
			const fromDocBlock = supportedBrowsers.find(
				( browser ) => browser.name().toLowerCase() === parsed.browser.toLowerCase()
			);

			if ( ! fromDocBlock ) {
				throw new Error( `Unsupported browser defined in DocBlock: ${ parsed.browser }` );
			}

			return fromDocBlock;
		}

		throw new Error( 'Unsupported DocBlock browser declaration.' );
```

Reply:
I accepted your suggestion but with minor changes. Thanks!